crates/swc_ecma_minifier/src/compress/optimize/params.rs

```diff
@@ -0,0 +1,363 @@
+            if scope_data.contains(ScopeData::USED_ARGUMENTS) {
+                return;
+            }
+        }
+
+        // Get usage info for the function
+        let usage = match self.data.vars.get(fn_id) {
+            Some(u) => u,
+            None => return,
+        };
+
+        // Skip if function is exported or can't optimize
+        if usage.flags.contains(VarUsageInfoFlags::EXPORTED)
+            || usage.flags.contains(VarUsageInfoFlags::INLINE_PREVENTED)
+            || usage.flags.contains(VarUsageInfoFlags::USED_AS_REF)
+            || usage.flags.contains(VarUsageInfoFlags::REASSIGNED)
+        {
+            return;
+        }
+
+        // Get call site arguments
+        let call_sites = match &usage.call_site_args {
+            Some(sites) if !sites.is_empty() => sites,
+            _ => return,
+        };
+
+        // Check if all call sites are known (callee_count should match call_sites
+        // length)
+        if usage.callee_count as usize != call_sites.len() {
+            return;
+        }
+
+        // Skip very small functions with few call sites that are better optimized
+        // by function body inlining (reduce_fns/reduce_vars) rather than parameter
+        // inlining.
+        //
+        // Small functions (single statement, simple return) with 1-2 calls are prime
+        // candidates for function inlining, which handles parameter substitution
+        // differently:
+        //   function g(b) { return b; }
+        //   g(2)
+        // =>
+        //   (function(b) { return 2; })(0)  // function inlined, param kept
+        //
+        // vs our parameter inlining which would do:
+        //   function g() { const b = 2; return b; }
+        //   g()  // param removed
+        //
+        // However, larger functions or functions with 3+ call sites benefit more from
+        // parameter inlining as function body inlining becomes less effective.
+        if (self.options.reduce_fns || self.options.reduce_vars) && call_sites.len() <= 2 {
+            // Check if this is a small, simple function (candidate for function
+            // inlining)
+            let is_small_function = if let Some(body) = &f.body {
+                // Single statement functions are prime candidates for function inlining
+                body.stmts.len() == 1
+            } else {
+                false
+            };
+
+            if is_small_function {
+                return;
+            }
+        }
+
+        // Analyze each parameter
+        let mut params_to_inline: Vec<(usize, Box<Expr>)> = Vec::new();
+
+        for (param_idx, param) in f.params.iter().enumerate() {
+            // Only handle simple identifier parameters
+            let param_id = match &param.pat {
+                Pat::Ident(ident) => ident.id.to_id(),
+                _ => continue, // Skip destructuring, rest params, etc.
+            };
+
+            // Check if parameter is mutated within the function
```

Comment:
im not sure this condition matters, it would just mean we need to inline as `let` vs `const`, but from a correctness perspective it isn't different i think